lib/phoenix_live_view/html_engine.ex

```diff
@@ -762,4 +765,50 @@ defmodule Phoenix.LiveView.HTMLEngine do
   defp void?(_), do: false
 
   defp to_location(%{line: line, column: column}), do: [line: line, column: column]
+
+  @attrs_with_required_ids ["phx-update", "phx-hook"]
+
+  # Check if attrs contains `@attrs_with_required_ids`.
+  #
+  # In case one is found and there is no ID, an error will be raised. Otherwise,
+  # it returns :ok.
+  defp validate_phx_attrs!(attrs, meta, state),
+    do: validate_phx_attrs!(attrs, meta, state, %{attr: nil, id?: false})
+
+  defp validate_phx_attrs!([], meta, state, %{attr: attr, id?: false})
+       when attr in @attrs_with_required_ids do
+    message = "attribute \"#{attr}\" requires the \"id\" attribute to be set"
+
+    raise ParseError,
+      line: meta.line,
+      column: meta.column,
+      file: state.file,
+      description: message
+  end
+
+  defp validate_phx_attrs!([], _meta, _state, _acc), do: :ok
+
+  defp validate_phx_attrs!([{"id", _} | t], meta, state, acc),
+    do: validate_phx_attrs!(t, meta, state, %{acc | id?: true})
+
+  defp validate_phx_attrs!([{"phx-update", {_type, value, _meta}} | t], meta, state, acc) do
+    if value in ~w(ignore append prepend) do
```

Comment:
@josevalim added the validation for `phx-update`. Let me know if that make sense. Do you see any other validation we should do?